src/main/java/com/openshift/internal/restclient/http/UrlConnectionHttpClient.java

```diff
@@ -295,6 +301,21 @@ private void setAcceptHeader(String acceptedVersion, String acceptedMediaType, H
 		connection.setRequestProperty(IHttpConstants.PROPERTY_ACCEPT, builder.toString());
 	}
 
+	private void setContentTypeHeader(String version, String mediaType, HttpURLConnection connection) {
+		if (StringUtils.isEmpty(mediaType)) {
+			throw new HttpClientException(MessageFormat.format(
+					"Accepted media type (ex. {0}) is not defined", IHttpConstants.MEDIATYPE_APPLICATION_JSON));
```

Comment:
Looks like we are allowing the mediatype to be set. We probably should not assume its JSON in the message